opus/application/apps/cart/views.py

```diff
@@ -673,62 +685,68 @@ def api_create_download(request, opus_id=None, fmt=None):
     hierarchical_struct = int(request.GET.get('hierarchical', 0))
     files_info = {}
     for f_opus_id in files:
-        if 'Current' not in files[f_opus_id]:
+        if downloadCurrentOnly and 'Current' not in files[f_opus_id]:
             continue
-        files_version = files[f_opus_id]['Current']
-        for product_type in files_version:
-            for file_data in files_version[product_type]:
-                path = file_data['path']
-                pretty_name = path.split('/')[-1]
-                logical_path = path[path.index('/holdings')+9:]
-                if pretty_name not in files_info:
-                    files_info[pretty_name] = [logical_path]
-                elif logical_path not in files_info[pretty_name]:
-                    files_info[pretty_name].append(logical_path)
+        for version_name in files[f_opus_id]:
+            if downloadCurrentOnly and version_name != 'Current':
+                continue
+            files_version = files[f_opus_id][version_name]
+            for product_type in files_version:
+                for file_data in files_version[product_type]:
+                    path = file_data['path']
+                    pretty_name = path.split('/')[-1]
+                    logical_path = path[path.index('/holdings')+9:]
```

Comment:
I'm not thrilled with hardcoding `/holdings` in the OPUS source code. That's really something that is the domain of PdsFile and OPUS shouldn't know about the directory structure. If we have to do this, let's at least put this string in settings.py.